cognite/powerops/resync/models/helpers.py

```diff
@@ -0,0 +1,133 @@
+import doctest
+import operator
+
+from functools import reduce
+from pprint import pformat
+from deepdiff.model import PrettyOrderedSet
+from typing import Any, Union, Type, get_args, get_origin
+from types import GenericAlias
+
+
+from cognite.client.data_classes import Relationship
+
+
+def isinstance_list(value: Any, type_: Type):
+    return isinstance(value, list) and value and isinstance(value[0], type_)
+
+
+def match_field_from_relationship(model_fields: list[str], relationship: Relationship) -> str:
```

Comment:
could benefit from a docstring, to understand what is does without having to read :)